core/src/main/java/org/infinispan/container/entries/ReadCommittedEntry.java

```diff
@@ -194,7 +195,9 @@ public final void commit(DataContainer container, EntryVersion newVersion) {
             if (isRemoved() && !isEvicted()) ahm.markRemoved(true);
          }
 
-         if (isRemoved()) {
+         if (isEvicted() && container instanceof DefaultDataContainer) {
```

Comment:
Trying to get a hold of the modeshape guys to find out if they need this or not.  If so I will add a EvictableDataContainer interface that defines this method.